synthesizer/src/store/helpers/memory/internal/mod.rs

```diff
@@ -37,8 +37,9 @@ pub struct MemoryMap<
     // with fast lookups and the keys could be typed (i.e. just `K` instead of `Vec<u8>`).
     map: Arc<RwLock<BTreeMap<Vec<u8>, V>>>,
     batch_in_progress: Arc<AtomicBool>,
-    atomic_batch: Arc<Mutex<IndexMap<K, Option<V>>>>,
+    atomic_batch: Arc<Mutex<Vec<(K, Option<V>)>>>,
     checkpoint: Arc<Mutex<VecDeque<usize>>>,
```

Comment:
this is not in this PR, but it seems to me that `checkpoint` could be a plain `Vec`